### PR TITLE
Cross-Platform Support for Admin Commands

### PR DESCRIPTION
**Pull request by gkahiu**
Minor changes to allow some of the commands in `admin.py` to be executed in a Windows setup.

### CODE REVIEW

File: admin.py [hunk 1]
```diff
@@ -274,7 +286,11 @@ def compile_resources(
     target_path = output_directory / "resources.py"
     target_path.parent.mkdir(parents=True, exist_ok=True)
     _log(f"compile_resources target_path: {target_path}", context=context)
-    subprocess.run(shlex.split(f"pyrcc5 -o {target_path} {resources_path}"))
+    subprocess.run(
+        shlex.split(
+            f"pyrcc5 -o {target_path.as_posix()} " f"{resources_path.as_posix()}"
```
Comment (Samweli) [inline admin.py:291]: The original path type is intended to work on all platflorms, what is the benefit of using a PosixPath.

**Review by Samweli — COMMENTED** [review]
@gkahiu Looks good, just one question about the usage of the PosixPath.

File: admin.py (hunk 1 above)
Comment (gkahiu) [reply admin.py:291]: Thanks Samweli. The paths use the default backslash in Windows and this does not work well with shlex.split() hence the need to force it to use posix path.

Comment (Samweli) [reply admin.py:291]: Okay, then we can check the platform os and then use the right path type, similar to what you have done when getting application data path for the QGIS profiles folder.

**Comment by gkahiu** [comment]
Hi @Samweli, is this PR still valid? Am assuming the `as_posix` function is valid across all platforms. I have tested on Windows and its working OK.

**Comment by Samweli** [comment]
> Hi @Samweli, is this PR still valid? Am assuming the `as_posix` function is valid across all platforms. I have tested on Windows and its working OK.

Hi @gkahiu yes it is, you just need to a check on the shlex.split function to only use the `as_posix` function when in Windows environment and leave the as it is for other environments.

**Comment by gkahiu** [comment]
Hi @Samweli, I have pushed the changes as proposed.

**Review by Samweli — APPROVED** [review]

**Pull request merged**